services/mbot_oled_display.py

```diff
@@ -190,14 +217,26 @@ def screen_services():
        
 
 def main():
+    if device is None or font is None or font_small is None:
+        logging.error("Initialization failed. Exiting application.")
+        return
+
+    logged_service_start = False
     while True:
-        screen_wifi()
-        time.sleep(SCREEN_CHANGE_DELAY)
-        screen_QR()
-        time.sleep(QR_SCREEN_CHANGE_DELAY)
-        screen_resources()
-        time.sleep(SCREEN_CHANGE_DELAY)
-        screen_services()
+        try:
+            screen_wifi()
+            time.sleep(SCREEN_CHANGE_DELAY)
+            screen_QR()
+            time.sleep(QR_SCREEN_CHANGE_DELAY)
+            screen_resources()
+            time.sleep(SCREEN_CHANGE_DELAY)
+            screen_services()
+            if not logged_service_start:
+                logging.info("OLED service started successfully.")
+                logged_service_start = True
+        except Exception as e:
+            logging.error(f"Unhandled exception during main loop: {e}")
+            break
```

Comment:
What are these errors likely to be? In general you want to catch specific exceptions and handle them as is appropriate based on the type. Depending on what kind of error arises, we make want to just log it and keep the script going, rather than quit the program.